src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -478,6 +487,71 @@ private function addServiceMethodCalls($id, Definition $definition, $variableNam
         return $calls;
     }
 
+    private function addServiceOverriddenGetters($id, Definition $definition)
+    {
+        if (PHP_VERSION_ID < 70000) {
+            throw new RuntimeException(sprintf('Getter-based injection requires PHP 7 or more as used for service "%s".', $id));
+        }
+        if (!isset($this->classResources[$class = $definition->getClass()])) {
+            $this->classResources[$class] = new \ReflectionClass($class);
+        }
+        $class = $this->classResources[$class];
+        if ($class->isFinal()) {
+            throw new RuntimeException(sprintf('Invalid getter for service "%s": class "%s" is final.', $id, $class->name));
+        }
+        $getters = '';
+        foreach ($definition->getOverriddenGetters() as $name => $returnValue) {
+            if (!$class->hasMethod($name)) {
+                throw new RuntimeException(sprintf('Invalid getter for service "%s": method "%s::%s" does not exist.', $id, $class->name, $name));
+            }
+            $r = $class->getMethod($name);
+            if ($r->isPrivate()) {
+                throw new RuntimeException(sprintf('Invalid getter for service "%s": method "%s::%s" is private.', $id, $class->name, $r->name));
+            }
+            if ($r->isStatic()) {
+                throw new RuntimeException(sprintf('Invalid getter for service "%s": method "%s::%s" is static.', $id, $class->name, $r->name));
+            }
+            if ($r->isFinal()) {
+                throw new RuntimeException(sprintf('Invalid getter for service "%s": method "%s::%s" is final.', $id, $class->name, $r->name));
+            }
+            if ($r->returnsReference()) {
+                throw new RuntimeException(sprintf('Invalid getter for service "%s": method "%s::%s" returns by reference.', $id, $class->name, $r->name));
+            }
+            if (0 < $r->getNumberOfParameters()) {
+                throw new RuntimeException(sprintf('Invalid getter for service "%s": method "%s::%s" has parameters.', $id, $class->name, $r->name));
```

Comment:
And maybe all of these can change in the beginning, to be clear this is coming from the service definition:

> Unable to configure getter injection for service "%s": ...